dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/WebSocket.war/src/io/openliberty/org/apache/faces40/fat/websocket/WebsocketPushBean.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.org.apache.faces40.fat.websocket;
+
+import java.io.Serializable;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.faces.event.WebsocketEvent;
+import jakarta.faces.event.WebsocketEvent.Closed;
+import jakarta.faces.event.WebsocketEvent.Opened;
+import jakarta.faces.push.Push;
+import jakarta.faces.push.PushContext;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+
+@Named
+@ApplicationScoped
+public class WebsocketPushBean implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private static Logger LOGGER = Logger.getLogger(WebsocketPushBean.class.getName());
+
+    @Inject
+    @Push
+    private PushContext myChannel;
+
+    public void send() {
+        LOGGER.log(Level.INFO, "Method send was invoked successfully!");
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            // TODO Auto-generated catch block
+            // Do you need FFDC here? Remember FFDC instrumentation and @FFDCIgnore
+            e.printStackTrace();
+        }
+//        throw new RuntimeException("Test onerror exception.");
```

Comment:
Already cleaned up,  will push in a new commit once a code review is completed.

Reply:
@nmittles  Can you explain as I don't follow this comment?  How will you test on-error in the final PR? Are you testing onerror via the RuntimeException or by the missing ENABLE_WEBSOCKET_ENDPOINT parameter? 

Also, I saw that close is called right after on error. Would you know if that's expected? I think so based on the existing behavior and what I've read.